src/core/qgsofflineediting.cpp

```diff
@@ -1796,19 +1635,24 @@ void QgsOfflineEditing::stopListenFeatureChanges()
               this, &QgsOfflineEditing::committedFeaturesRemoved );
 }
 
-void QgsOfflineEditing::layerAdded( QgsMapLayer *layer )
+void QgsOfflineEditing::setupLayer( QgsMapLayer *layer )
 {
   Q_ASSERT( layer );
 
-  // detect offline layer
-  if ( layer->customProperty( CUSTOM_PROPERTY_IS_OFFLINE_EDITABLE, false ).toBool() )
+  QgsVectorLayer *vLayer = qobject_cast<QgsVectorLayer *>( layer );
+  if ( vLayer )
   {
-    QgsVectorLayer *vLayer = qobject_cast<QgsVectorLayer *>( layer );
-
-    Q_ASSERT( vLayer );
-
-    connect( vLayer, &QgsVectorLayer::editingStarted, this, &QgsOfflineEditing::startListenFeatureChanges );
-    connect( vLayer, &QgsVectorLayer::editingStopped, this, &QgsOfflineEditing::stopListenFeatureChanges );
+    // detect offline layer
+    if ( vLayer->customProperty( CUSTOM_PROPERTY_IS_OFFLINE_EDITABLE, false ).toBool() )
+    {
+      connect( vLayer, &QgsVectorLayer::editingStarted, this, &QgsOfflineEditing::startListenFeatureChanges );
+      connect( vLayer, &QgsVectorLayer::editingStopped, this, &QgsOfflineEditing::stopListenFeatureChanges );
+    }
+    else
+    {
+      disconnect( vLayer, &QgsVectorLayer::editingStarted, this, &QgsOfflineEditing::startListenFeatureChanges );
```

Comment:
For most layers added (those without the CUSTOM_PROPERTY_IS_OFFLINE_EDITABLE set to true), these connections will never exist in the first place, so disconnecting them will raise a qt warning (i think)